### PR TITLE
Add header text align theme variable to stack cards

### PR DESCRIPTION
**Pull request by dhoeben**
## Proposed change

I added a variable to hui-stack, which will impact `horizontal-stack`, `vertical-stack` and `grid`. This way, when defining a variable in the theme, users are able to align their title in the way they want. I also added a fallback, to align the title at the start, as it is now.

Info: https://developer.mozilla.org/en-US/docs/Web/CSS/text-align

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
theme:
  ha-card-header-text-align: center
```

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/cards/hui-stack-card.ts [hunk 1]
```diff
@@ -92,6 +92,7 @@ export abstract class HuiStackCard<T extends StackCardConfig = StackCardConfig>
     return css`
       .card-header {
         color: var(--ha-card-header-color, var(--primary-text-color));
+        text-align: var(--ha-card-header-text-align, start);
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-stack-card.ts:95]: The variable sounds like it is for `ha-card` but you implemented it just for stack cards. I would expect this to be at least also implemented in `ha-card`.

**Comment by dhoeben** [comment]
The variable is in line with the rest of the variable names used in the hui-stack. If you don't like the variable names, I can change all of them if you like!

**Comment by bramkragten** [comment]
> The variable is in line with the rest of the variable names used in the hui-stack. If you don't like the variable names, I can change all of them if you like!

Thats is because it is using the variables of `ha-card`, because it uses the same style as `ha-card`. You should definitely not change those.

**Comment by dhoeben** [comment]
So you are saying that you don't want to add the possibility to align titles of the vertical stack card, horizontal stack card and grid card?

**Comment by bramkragten** [comment]
> So you are saying that you don't want to add the possibility to align titles of the vertical stack card, horizontal stack card and grid card?

No? I'm just saying you should add to more places if we are going to add it, to make it consistent.

**Comment by dhoeben** [comment]
The difference is that with the hui-stack, the title is placed "outside" the card. And with other cards it is located "inside" the card. As titles are different for stacks then for cards.

File: src/panels/lovelace/cards/hui-stack-card.ts (hunk 1 above)
Comment (dhoeben) [reply src/panels/lovelace/cards/hui-stack-card.ts]: Better like this?

File: src/panels/lovelace/cards/hui-stack-card.ts [hunk 2]
```diff
@@ -92,6 +92,7 @@ export abstract class HuiStackCard<T extends StackCardConfig = StackCardConfig>
     return css`
       .card-header {
         color: var(--ha-card-header-color, var(--primary-text-color));
+        text-align: var(--hui-stack-title-text-align, start);
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-stack-card.ts:95]:
```suggestion
        text-align: var(--ha-stack-title-text-align, start);
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**